internal/proof/proof.go

```diff
@@ -22,23 +22,39 @@ const (
 	proofTxTimeout uint64 = 60
 )
 
+// ProofchainInteractor a wrapper over proofchain contract to help clients interact with it
+type ProofchainInteractor struct {
+	config             *config.AgentConfig
+	ethClient          *ethclient.Client
+	proofChainContract *ProofChain
+}
+
+// NewProofchainInteractor creates a new `ProofchainInteractor` and does the setup
+func NewProofchainInteractor(config *config.AgentConfig, ethClient *ethclient.Client) *ProofchainInteractor {
+	interactor := &ProofchainInteractor{config: config, ethClient: ethClient}
+	contractAddress := common.HexToAddress(config.ProofchainConfig.ProofChainAddr)
+	contract, err := NewProofChain(contractAddress, ethClient)
```

Comment:
i assume proofchain contract can be reused across multiple SendBlockReplicaProofTx calls.